sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/proxy/ApiClientInvocationHandler.java

```diff
@@ -39,12 +60,79 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
     } catch (InvocationTargetException invocationTargetException) {
       Throwable cause = invocationTargetException.getCause();
       if (cause instanceof WebApplicationException) {
-        throw new ClientRequestException(cause);
+        throwIfNotSuppressed(method, new ClientRequestException(cause));
       } else if (cause instanceof ProcessingException) {
-        throw new ClientRequestException(cause);
+        throwIfNotSuppressed(method, new ClientRequestException(cause));
       } else {
         throw cause;
       }
+      LOG.info(
+          "API client error '{}' suppressed to null when calling {}", cause.getMessage(), method);
+      return null;
     }
   }
+
+  /**
+   * @throws ClientRequestException the given {@code exceptionToThrow} if not suppressed by
+   *     annotations of the {@code invokedMethod}.
+   */
+  private void throwIfNotSuppressed(Method invokedMethod, ClientRequestException exceptionToThrow) {
+    Optional<Integer> httpStatus = getHttpStatus(exceptionToThrow);
+    if (httpStatus.isPresent()) {
+      if (!isHttpStatusSuppressed(invokedMethod, httpStatus.get())) {
+        throw exceptionToThrow;
+      }
+    } else if (exceptionToThrow.isConnectTimeout()) {
+      if (!isConnectTimeoutErrorSuppressed(invokedMethod)) {
+        throw exceptionToThrow;
+      }
+    } else if (exceptionToThrow.isReadTimeout()) {
+      if (!isReadTimeoutErrorSuppressed(invokedMethod)) {
+        throw exceptionToThrow;
+      }
+    } else if (exceptionToThrow.isProcessingError()) {
+      if (!isProcessingErrorSuppressed(invokedMethod)) {
+        throw exceptionToThrow;
+      }
+    } else {
+      throw exceptionToThrow;
+    }
+    exceptionToThrow.close();
+  }
+
+  private Optional<Integer> getHttpStatus(ClientRequestException exceptionToThrow) {
+    return exceptionToThrow.getResponse().map(Response::getStatus);
+  }
+
+  private boolean isHttpStatusSuppressed(Method invokedMethod, int httStatusCode) {
+    SuppressHttpErrorsToNull suppressedHttpErrors =
+        invokedMethod.getDeclaredAnnotation(SuppressHttpErrorsToNull.class);
+    if (suppressedHttpErrors == null) {
+      return false;
+    }
+    Set<Integer> suppressedStatusCodes =
+        IntStream.of(suppressedHttpErrors.value()).boxed().collect(Collectors.toSet());
+    if (suppressedHttpErrors.allRedirectErrors()) {
+      suppressedStatusCodes.addAll(ALL_REDIRECT_ERRORS);
+    }
+    if (suppressedHttpErrors.allClientErrors()) {
+      suppressedStatusCodes.addAll(ALL_CLIENT_ERRORS);
+    }
+    if (suppressedHttpErrors.allServerErrors()) {
+      suppressedStatusCodes.addAll(ALL_SERVER_ERRORS);
+    }
+    return suppressedStatusCodes.contains(httStatusCode);
+  }
+
+  private boolean isProcessingErrorSuppressed(Method invokedMethod) {
+    return invokedMethod.getDeclaredAnnotation(SuppressProcessingErrorsToNull.class) != null;
```

Comment:
For readability I prefer `isAnnotationPresent`

Reply:
Fixed.